app/controllers/reports_controller.rb

```diff
@@ -31,8 +31,61 @@ def identifiers_to_reconcile
     end
   end
 
+  def pulfa_ark_report
+    authorize! :show, Report
+    if params[:since_date]
+      @resources = ark_report_resources(params[:since_date]) || []
+    end
+
+    respond_to do |format|
+      format.html
+      format.csv do
+        send_data hashes_to_csv(["id", "component_id", "ark", "url"], @resources.map { |r| resource_hash(r) }),
+                  filename: "pulfa-ark-report-#{params[:since_date]}-to-#{Time.zone.today}.csv"
+      end
+    end
+  end
+
   private
 
+    def ark_report_resources(since_date)
```

Comment:
Maybe we should extract this to a query, so we have somewhere to make this more efficient eventually?

Reply:
👍 Yes, I've pulled that out to a custom query and switched to doing it with SQL, which will surely perform much, much better: https://github.com/pulibrary/figgy/pull/2606/files#diff-4a0243fdb3248f71bcb779a46ce59511